libsolidity/ast/Types.cpp

```diff
@@ -617,6 +617,21 @@ FixedPointType::FixedPointType(unsigned _totalBits, unsigned _fractionalDigits,
 		"Invalid bit number(s) for fixed type: " +
 		dev::toString(_totalBits) + "x" + dev::toString(_fractionalDigits)
 	);
+
+	solAssert(
+		isValid(_totalBits, _fractionalDigits, isSigned()),
+		"Number of fractional digits do not fit the specified bit width: " +
+		dev::toString(_totalBits) + "x" + dev::toString(_fractionalDigits)
+	);
+}
+
+bool FixedPointType::isValid(unsigned _totalBits, unsigned _fractionalDigits, bool _isSigned)
+{
+	if (_totalBits < 8 || _totalBits > 256 || _totalBits % 8 != 0 || _fractionalDigits > 80)
+		return false;
+
+	unsigned const bitsAvailable = _totalBits - (_isSigned ? 1 : 0);
+	return boost::multiprecision::pow(bigint(10), _fractionalDigits) <= (bigint(1) << bitsAvailable);
```

Comment:
What exactly does this check? That `1.0` is representable? Why is it `<=` and not `<`?

Reply:
@chriseth I think it checks whether the number of fractional decimal digits is small enough to actually fit into the total number of bits.
The idea is that a type like ``fixed1x2`` doesn't make much sense or is at least confusing, since two decimal digits can't be represented in one bit. If ``10^_fractionalDigits <= 2^bitsAvailable``, then at least the number of decimal digits supposed to be representable in the type is actually representable in the type.

If, on the other hand, you think of ``ufixedXxY`` as a type with ``X`` bits that is implicitly multiplied by ``10^(-Y)``, then this is the same as checking whether ``1.0`` is representable.
Arguably one could do the opposite and allow a ``fixed1x2`` type, if one thinks about them this way. To be honest I find the current naming of fixed point values as ``X`` being total bits and ``Y`` being decimal places pretty strange in general, regardless of how one deals with cases like this.

In any case: if one wants this check, I think ``<=`` makes more sense than ``<``.

Reply:
Well it is mathematically impossible for them to be equal, so it doesn't really matter at all.

Proof: 10 is divisible by 5, so (10^n1) is also divisible by five. 2 is not divisible by 5, so (2^n2) is not divisible by 5. It is impossible for a number divisible by 5 to be equal to one that is not divisible by 5. QED.

Reply:
Actually, ``<=`` and ``<`` will even be the same in this case, since ``10^x`` will never actually be equal to ``2^y``, unless ``x`` and ``y`` are zero and ``_totalBits`` is already required to be at least ``8``.

Reply:
@meowingtwurtle Yep, that's what I just noticed as well :).

Reply:
Could you please add a comment? I think it is better to explain it using `1.0 is representable`, since the sentence `whether the number of fractional decimal digits is small enough to actually fit into the total number of bits.` still lacks a definition of what it means for fractional digits to fit into the total number of bits.